src/components/views/messages/DateSeparator.tsx

```diff
@@ -132,28 +135,106 @@ export default class DateSeparator extends React.Component<IProps, IState> {
+            // about after a slow request if they've already navigated away to another
+            // room.
+            const currentRoomId = SdkContextClass.instance.roomViewStore.getRoomId();
+            if (currentRoomId === roomIdForJumpRequest) {
+                let friendlyErrorMessage = `An error occured while trying to find and jump to the given date.`;
+                let submitDebugLogsContent: JSX.Element = <></>;
+                if (err?.name === "ConnectionError" || err?.httpStatus) {
+                    if (err?.errcode === "M_NOT_FOUND") {
+                        friendlyErrorMessage = _t(
+                            "We were unable to find an event looking forwards from %(dateString)s. " +
+                                "Try choosing an earlier date.",
+                            { dateString: formatFullDateNoDay(new Date(unixTimestamp)) },
+                        );
+                    } else {
+                        friendlyErrorMessage = _t(
+                            "A network error occurred while trying to find and jump to the given date. " +
+                                "Your homeserver might be down or was just a temporary problem with your " +
+                                "internet connection. Please try again. If this continues, please " +
+                                "contact your homeserver administrator.",
+                        );
+                    }
+                } else {
+                    // We only give the option to submit logs for actual errors, not network problems.
+                    submitDebugLogsContent = (
+                        <p>
+                            {_t(
+                                "Please submit <debugLogsLink>debug logs</debugLogsLink> to help us " +
+                                    "track down the problem.",
+                                {},
+                                {
+                                    debugLogsLink: (sub) => (
+                                        <AccessibleButton
+                                            // This is by default a `<div>` which we
+                                            // can't nest within a `<p>` here so update
+                                            // this to a be a inline anchor element.
+                                            element="a"
+                                            kind="link"
+                                            onClick={() => this.onBugReport(err)}
+                                            data-testid="jump-to-date-error-submit-debug-logs-button"
+                                        >
+                                            {sub}
+                                        </AccessibleButton>
+                                    ),
+                                },
+                            )}
+                        </p>
+                    );
+                }
+
                 Modal.createDialog(ErrorDialog, {
```

Comment:
I don't know how to fix this TypeScript error:

```
src/components/views/messages/DateSeparator.tsx:214:36 - Argument of type 'typeof ErrorDialog' is not assignable to parameter of type 'ComponentType'.
```

Reply:
@alunturner Any tips on how to fix this error?

Reply:
I'm afraid I think I've probably done the same as you and checked the other uses of this, seeing that they all have the same issue. The problem must lie in `Modal.createDialog` (so it seems to me) given the consistency of the TS errors throughout the codebase. If I had tips, I'd give them.